pkg/kubectl/resource/builder_test.go

```diff
@@ -234,32 +234,73 @@ func TestNodeBuilder(t *testing.T) {
 }
 
 func TestPathBuilderWithMultiple(t *testing.T) {
+	// This builder *should not* error
 	b := NewBuilder(testapi.Default.RESTMapper(), api.Scheme, fakeClient(), testapi.Default.Codec()).
-		FilenameParam(false, "../../../examples/guestbook/redis-master-controller.yaml").
-		FilenameParam(false, "../../../examples/pod").
+		FilenameParam(false, false, "../../../examples/pod").
+		FilenameParam(false, true, "../../../examples/recursive/pod").
+		FilenameParam(false, false, "../../../examples/guestbook/redis-master-controller.yaml").
+		FilenameParam(false, true, "../../../examples/recursive/rc").
+		FilenameParam(false, false, "../../../examples/inode/hardlink/busybox-link.yaml").
+		FilenameParam(false, true, "../../../examples/inode/hardlink/busybox-link.yaml").
 		NamespaceParam("test").DefaultNamespace()
 
 	test := &testVisitor{}
 	singular := false
 
 	err := b.Do().IntoSingular(&singular).Visit(test.Handle)
-	if err != nil || singular || len(test.Infos) != 2 {
+	if err != nil || singular || len(test.Infos) != 12 {
 		t.Fatalf("unexpected response: %v %t %#v", err, singular, test.Infos)
 	}
 
 	info := test.Infos[0]
+	if _, ok := info.Object.(*api.Pod); !ok || info.Name != "nginx" || info.Namespace != "test" {
+		t.Errorf("unexpected info: %#v", info)
+	}
+	for i, v := range []int{1, 2, 3, 4} { // 4 tests as there are 4 resources in ../../../examples/recursive/pod
+		info = test.Infos[v]
+		if _, ok := info.Object.(*api.Pod); !ok || info.Name != fmt.Sprintf("busybox%d", i) || info.Namespace != "test" {
+			t.Errorf("unexpected info: %#v", info)
+		}
+	}
+	info = test.Infos[5]
 	if _, ok := info.Object.(*api.ReplicationController); !ok || info.Name != "redis-master" || info.Namespace != "test" {
 		t.Errorf("unexpected info: %#v", info)
 	}
-	info = test.Infos[1]
-	if _, ok := info.Object.(*api.Pod); !ok || info.Name != "nginx" || info.Namespace != "test" {
+	for i, v := range []int{6, 7, 8, 9} { // 4 tests as there are 4 resources in ../../../examples/recursive/rc
+		info = test.Infos[v]
+		if _, ok := info.Object.(*api.ReplicationController); !ok || info.Name != fmt.Sprintf("busybox%d", i) || info.Namespace != "test" {
+			t.Errorf("unexpected info: %#v", info)
+		}
+	}
+	info = test.Infos[10]
+	if _, ok := info.Object.(*api.Pod); !ok || info.Name != "busybox" || info.Namespace != "test" {
 		t.Errorf("unexpected info: %#v", info)
 	}
+	info = test.Infos[11]
+	if _, ok := info.Object.(*api.Pod); !ok || info.Name != "busybox" || info.Namespace != "test" {
+		t.Errorf("unexpected info: %#v", info)
+	}
+
+	// This builder *should* error
+	b = NewBuilder(testapi.Default.RESTMapper(), api.Scheme, fakeClient(), testapi.Default.Codec()).
```

Comment:
Separate test for these please.

Also, checking symlinks into the repo can cause problems for some tooling (people have yelled at me in other repos).  It sounds ugly, but constructing the borked structure inside your test, inside of a `/tmp` dir will keep the bad structures contained, but still allow us to run the test.  And don't forget to clean them up.